### ISSUE TITLE
Hotring fastest lap time always recorded in stats with fractional part set to 00

### ISSUE THREAD

**Issue by Barracuda6** [GTAmodding/re3#1215]
**Describe the bug**
Upon completion of the Hotring stadium event, your best lap time is displayed on screen to two decimal places (e.g. 15.57). On the stats screen, however, the "Fastest lap time in Hotring" stat is always recorded with the fractional part as "00" (e.g. 15.00).

**To Reproduce**
1. Complete the Hotring stadium event and record the best lap time that is displayed on screen
2. Check the "Fastest lap time in Hotring" stat on the stats screen and compare the value to what was displayed on screen at the end of the Hotring event

**Expected behavior**
The fastest lap time on the stats screen should display the same lap time displayed on screen at the end of the Hotring event. This issue is also present in vanilla.

**Screenshots**

![screen_0 1625189857](https://user-images.githubusercontent.com/41495078/125141666-324e2400-e105-11eb-92ed-0d2705b08b29.png)

![screen_01625504225](https://user-images.githubusercontent.com/41495078/125141679-3c702280-e105-11eb-8519-dd279f37ddd7.png)

**Version**
Event was completed on Linux 64-bit OpenGL OAL a0c76102 (https://github.com/withmorten/re3/commit/a0c76102)

**Comment by Sergeanur** [comment]
Is it vanilla bug?

**Comment by Barracuda6** [comment]
> Is it vanilla bug?

Yes, this happens in vanilla too

**Comment by withmorten** [comment]
For reference, `STAT_LINE_1(float, "STFT_22", FastestTimes[21] / 1000.0f, 1);` shows the fractional, but it shows it as 15.58

**Comment by Barracuda6** [comment]
This line was added to the end of CStats::LoadStats in src/core/Stats.cpp to print the "fastest lap time" value stored in the save file:

```c++
printf("FastestTimes[21] is %08X\n", FastestTimes[21]);
```

The value returned was 00003CD8 (15576).

This is the save file used for the screenshots in this issue and also the calculations below: 
[GTAVCsf6_hotring_completed.b.zip](https://github.com/GTAmodding/re3/files/6794486/GTAVCsf6_hotring_completed.b.zip)

---

Snippet from main.scm showing the part that calculates the value to display on screen as the fastest lap time at the end of the Hotring event:

```
:OVALRNG_16506
0084: $7458 = $1594 // $ = $ int    $1594 = 15576 (fastest lap time)
$7458 /= 1000 // $ /= int           $7458 = 15576 / 1000 = 15.576 = 15 because int? (FINAL VALUE)
0084: $7003 = $7458 // $ = $ int    $7003 = 15 because int?
$7003 *= 1000 // $ *= int           $7003 = 15 * 1000 = 15000
0084: $7451 = $1594 // $ = $ int    $7451 = 15576
0060: $7451 -= $7003 // $ -= $ int  $7451 = 15576 - 15000 = 576
0084: $7003 = $7451 // $ = $ int    $7003 = 576
$7003 /= 10 // $ /= int             $7003 = 576 / 10 = 57.6 = 57 because int? (FINAL VALUE)
if 
  $7003 > 9 // $ > int 
jf @OVALRNG_16653 
045B: text_draw_2numbers 32.0 332.0 'HOTR_27' $7458 $7003  // 15 . 57 // [HOTR_27:OVALRIG] = Best Lap Time: ~1~.~1~ seconds
gosub @OVALRNG_17584 
045B: text_draw_2numbers 30.0 330.0 'HOTR_27' $7458 $7003 
jump @OVALRNG_16712 

:OVALRNG_16653
045B: text_draw_2numbers 32.0 332.0 'HOTR_28' $7458 $7003  // 15 . 57 // [HOTR_28:OVALRIG] = Best Lap Time: ~1~.0~1~ seconds
gosub @OVALRNG_17584 
045B: text_draw_2numbers 30.0 330.0 'HOTR_28' $7458 $7003 
```

**Comment by MBH69** [comment]
NY

**Comment by Barracuda6** [comment]
> For reference, `STAT_LINE_1(float, "STFT_22", FastestTimes[21] / 1000.0f, 1);` shows the fractional, but it shows it as 15.58

`STAT_LINE_1(float, "STFT_22", floor(FastestTimes[21] / 10) / 100, 1);` shows it as 15.57, not sure if there's a better way to do it:

![screen_01625929355](https://user-images.githubusercontent.com/41495078/125167971-bd293000-e192-11eb-8d82-2fdb01694b87.png)

**Comment by withmorten** [comment]
Fixed in 8a114514d9ea61815c2d391bc5edaa69d1648bfe
